ui/scripts/modal.js

```diff
@@ -1,18 +1,29 @@
 export default class Modal {
     // @todo: will need to add trailer url later
-    constructor(title) {
+    constructor(title, trailer_url) {
         this.title = title;
-        // this.url = url;
+        this.trailer_url = trailer_url;
+    }
+
+    getId(url) {
+        const regExp = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/;
```

Comment:
Why is the regex so complex? I don't think I pass youtube links in this format from the backend. (E.g. it doesn't use "shortened" domains like `youtu.be`.) The actual example is `https://www.youtube.com/watch?v=JJZzlaH9ECc`, and that's what you should expect (and nothing else - for youtube at least; and we don't have anything else.)

Overall, I suspect we could live without regex here at all; just check that url starts with "https://www.youtube.com/watch?v=" and return the rest. What's the drawback?

The reason I think it's better to avoid regexes is because they are more expensive than simple string operations like searching for a substring.

Reply:
Thanks for the tip, I sure overcomplicated it.